lib/knapsack_pro/repository_adapters/git_adapter.rb

```diff
@@ -44,25 +44,43 @@ def build_author
 
       def git_commit_authors
         if KnapsackPro::Config::Env.ci? && shallow_repository?
-          command = 'git fetch --shallow-since "one month ago" --quiet 2>/dev/null'
+          command = 'git fetch --shallow-since "one month ago" --quiet'
           begin
             Timeout.timeout(5) do
-              `#{command}`
+              Kernel.system(command, out: File::NULL, err: File::NULL)
             end
           rescue Timeout::Error
             KnapsackPro.logger.debug("Skip the `#{command}` command because it took too long.")
           end
         end
 
-        `git log --since "one month ago" 2>/dev/null | git shortlog --summary --email 2>/dev/null`
+        summary_read, log_write = IO.pipe
+        Kernel.system('git log --since "one month ago"', out: log_write, err: File::NULL)
+        log_write.close
+        summary, summary_write = IO.pipe
+        Kernel.system('git shortlog --summary --email', in: summary_read, out: summary_write, err: File::NULL)
+        summary_read.close
+        summary_write.close
+        result = summary.read
+        summary.close
+        result
```

Comment:
recommendation

This should work:

```suggestion
        IO.pipe do |log_r, log_w|
          Kernel.system('git log --since "one month ago"', out: log_w, err: File::NULL)
          log_w.close
          IO.pipe do |shortlog_r, shortlog_w|
            Kernel.system('git shortlog --summary --email', in: log_r, out: shortlog_w, err: File::NULL)
            shortlog_w.close
            shortlog_r.read
          end
        end
```